src/pipelines.py

```diff
@@ -301,6 +303,96 @@ def mask_postprocessing(model, config, make_transformer, **kwargs):
     return score_builder
 
 
+def lgbm_train(config):
+    save_output = False
+    unet_type = 'weighted'
+    config['execution']['stream_mode'] = True
+
+    if unet_type == 'standard':
+        unet_pipeline = unet(config, train_mode=False)
+    elif unet_type == 'weighted':
+        unet_pipeline = unet_weighted(config, train_mode=False)
+    else:
+        raise NotImplementedError
+
+    mask_dilation = unet_pipeline.get_step('mask_dilation')
+    mask_resize = unet_pipeline.get_step('mask_resize')
+
+    feature_extractor = Step(name='feature_extractor',
+                             transformer=post.FeatureExtractor(**config['postprocessor']['feature_extractor']),
+                             input_steps=[mask_dilation, mask_resize],
+                             input_data=['input'],
+                             adapter={'images': ([(mask_dilation.name, 'dilated_images')]),
+                                      'probabilities': ([(mask_resize.name, 'resized_images')]),
+                                      'annotations': ([('input', 'annotations')]),
+                                      },
+                             cache_dirpath=config.env.cache_dirpath,
+                             save_output=save_output)
+
+    scoring_model = Step(name='scoring_model',
+                         transformer=ScoringLightGBM(**config['postprocessor']['lightGBM'])
+                         if config['postprocessor']['scoring_model'] == 'lgbm' else
+                         ScoringRandomForest(**config['postprocessor']['random_forest']),
+                         input_steps=[feature_extractor],
+                         cache_dirpath=config.env.cache_dirpath,
+                         save_output=save_output,
+                         is_trainable=True,
+                         force_fitting=False
+                         )
+
+    return scoring_model
+
+
+def lgbm_inference(config, input_pipeline):
```

Comment:
@jakubczakon OK, thx

Reply:
@jakubczakon I think we need `config` to be the first argument, because in pipeline_manager.py we call `pipeline(SOLUTION_CONFIG)`, not `pipeline(config=SOLUTION_CONFIG)`, so it tries to set first argument of `lgbm_inference` to SOLUTION_CONFIG.
I have to keep the order, or change pipeline_manager.py